### ISSUE TITLE
Game crashes upon loading level.

### ISSUE THREAD

**Issue by PRYZMwubs** [truong-bui/AsyncLoadingScreen#48]
Hi there, our game crashes upon trying to load a level. It doesn't crash when the plugin is disabled. I'm using the marketplace build. The plugin worked fine until recently. This was on UE 4.27.

The log is as follows:

"LogPlayLevel: Error: [2022.01.24-02.41.07:457][727]LogThreadingWindows: Error: Runnable thread RHIThread crashed.
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: begin: stack for UAT
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: === Critical error: ===
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error:
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: Fatal error!
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error:
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0xffffffffffffffff
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error:
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e4d7f15c Mirem.exe!FD3D12CommandContext::SetRenderTargets() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e4dffab4 Mirem.exe!FD3D12CommandContextBase::RHIBeginDrawingViewport() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e66e32f7 Mirem.exe!FRHICommand<FRHICommandBeginDrawingViewport,FRHICommandBeginDrawingViewportString1871>::ExecuteAndDestruct() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e66e6193 Mirem.exe!FRHICommandListExecutor::ExecuteInner_DoExecute() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e66d627f Mirem.exe!FExecuteRHIThreadTask::DoTask() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e66e8164 Mirem.exe!TGraphTask<FExecuteRHIThreadTask>::ExecuteTask() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e3a9515b Mirem.exe!FNamedTaskThread::ProcessTasksNamedThread() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e3a955be Mirem.exe!FNamedTaskThread::ProcessTasksUntilQuit() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e67a69ad Mirem.exe!FRHIThread::Run() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e3fb298f Mirem.exe!FRunnableThreadWin::Run() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff7e3fa8ea0 Mirem.exe!FRunnableThreadWin::GuardedRun() []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff9758a7034 KERNEL32.DLL!UnknownFunction []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: [Callstack] 0x00007ff9765c2651 ntdll.dll!UnknownFunction []
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error:
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: Crash in runnable thread RHIThread
LogPlayLevel: Error: [2022.01.24-02.41.07:458][727]LogWindows: Error: end: stack for UAT
LogPlayLevel: Error: ERROR: Client exited with error code: 3"

**Comment by PRYZMwubs** [comment]
UPDATE: Tried the github branch and I get the same problem. It works fine when loading the main menu but when trying to load any other level it crashes with Unreal stating "fatal error."
![Screenshot (32)](https://user-images.githubusercontent.com/47482618/150714117-f9af7442-720f-4809-9fd7-641362c6c640.png)

**Comment by truong-bui** [comment]
I don't think this relates to the Async Loading Screen plugin. Try to disable the plugin and Launch again to see if there is any crash. Also, delete Binaries/Interrmeridate/Saved folders then try again.

**Comment by PRYZMwubs** [comment]
> I don't think this relates to the Async Loading Screen plugin. Try to disable the plugin and Launch again to see if there is any crash. Also, delete Binaries/Interrmeridate/Saved folders then try again.

I already tried that. The game loads correctly without the plugin. It is 100% a problem with the plugin.

**Comment by PRYZMwubs** [comment]
UPDATE: Just to be thorough, I did delete the Binaries, Intermediate and Saved folders and I still get the same problem.

**Comment by truong-bui** [comment]
Can you run this plugin in a new project on your PC to see if this bug happens again? Sorry, I can't help you if I can't reproduce this issue on my side. As I've researched, it may relate to unsupported renderer API. Try to change to Vulkan, DirectX11, OpenGL.

**Comment by LingHyun** [comment]
I had the same problem
